generate/generate.go

```diff
@@ -222,94 +200,222 @@ func Do(rootdir string, workingDir string) Report {
 		return report
 	})
 
-	outdatedDirs, err := listGenFilesOutsideStacks(rootdir, rootdir)
+	orphanedGenFiles, err := ListGenFiles(rootdir, rootdir)
 	if err != nil {
 		report.CleanupErr = err
 		return report
 	}
 
-outdatedDirsLoop:
-	for _, outdatedDir := range outdatedDirs {
-		relpath := project.NewPath(filepath.ToSlash(strings.TrimPrefix(outdatedDir.dir, rootdir)))
-		deleted := []string{}
-
-		for _, outdatedFile := range outdatedDir.files {
-			if err := os.Remove(filepath.Join(outdatedDir.dir, outdatedFile)); err != nil {
-				report.Failures = append(report.Failures, FailureResult{
-					Result: Result{
-						Dir:     relpath,
-						Deleted: deleted,
-					},
-					Error: err,
-				})
-				continue outdatedDirsLoop
+	deletedFiles := map[project.Path][]string{}
+	deleteFailures := map[project.Path]*errors.List{}
```

Comment:
Since these are not stacks each dir is reported independently, so we need to organize the report per directory here. Was on the fence between doing this on the report itself with a method or doing it here, in the end did it here with these maps.